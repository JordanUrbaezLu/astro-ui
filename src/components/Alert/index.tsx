/* istanbul ignore file */
export * from "./Alert";
