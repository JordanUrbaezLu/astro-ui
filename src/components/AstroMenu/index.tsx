export * from "./AstroMenu";
export * from "./AstroMenuItem";
