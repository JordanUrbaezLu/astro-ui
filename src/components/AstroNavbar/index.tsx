export * from "./AstroNavbar";
export * from "./AstroNavbarItem";
