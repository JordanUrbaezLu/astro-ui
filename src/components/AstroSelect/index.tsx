export * from "./AstroSelect";
export * from "./AstroSelectItem";
