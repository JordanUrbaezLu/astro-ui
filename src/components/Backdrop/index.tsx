/* istanbul ignore file */
export * from "./Backdrop";
