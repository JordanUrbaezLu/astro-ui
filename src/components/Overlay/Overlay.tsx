import classNames from "classnames";
import * as React from "react";
import styles from "./Overlay.module.scss";

export interface OverlayProps
  extends React.ComponentPropsWithoutRef<"div"> {
  /**
   * The content for the Overlay
   */
  children?: React.ReactNode;
  /**
   * If the Overlay shows the background
   *
   * @default true
   */
  showBackground?: boolean;
}

/**
 * @utility
 */
export const Overlay: React.FC<OverlayProps> = (props) => {
  const {
    className,
    children,
    showBackground = true,
    ...rest
  } = props;

  return (
    <div
      className={classNames(
        showBackground ? styles.overlay : styles.noBackground,
        className
      )}
      {...rest}
    >
      {children}
    </div>
  );
};
