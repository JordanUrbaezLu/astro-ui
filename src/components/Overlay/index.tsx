/* istanbul ignore file */
export * from "./Overlay";
