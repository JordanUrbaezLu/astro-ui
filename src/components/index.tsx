export * from "./Alert";
export * from "./Button";
export * from "./Checkbox";
export * from "./DatePicker";
export * from "./Link";
export * from "./Menu";
export * from "./Tooltip";
export * from "./Select";
export * from "./Switch";
export * from "./TextField";
