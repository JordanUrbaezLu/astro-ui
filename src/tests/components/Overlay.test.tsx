import { screen, render } from "@testing-library/react";
import { axe } from "jest-axe";
import { Overlay } from "../../components/Overlay";

describe("Overlay", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Overlay />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(<Overlay />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render with no background correctly.", () => {
    render(<Overlay data-testid="overlay" showBackground={false} />);
    expect(screen.getByTestId("overlay")).toHaveClass("noBackground");
  });
});
