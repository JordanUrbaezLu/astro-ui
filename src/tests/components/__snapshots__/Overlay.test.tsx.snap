// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Should render component correctly. 1`] = `
<div
  class="overlay"
/>
`;
